### PR TITLE
feat(provider/kubernetes): v2 manifest provider

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/Manifest.java [hunk 1]
```diff
@@ -17,8 +17,8 @@
 
 package com.netflix.spinnaker.clouddriver.model;
 
-public interface Manifest<S> {
-  S getStatus();
-  boolean isReady();
+public interface Manifest {
+  boolean isStable();
```
Comment (danielpeach) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/Manifest.java]: What does this signify?

Comment (lwander) [reply clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/Manifest.java]: That the manifest has reached it desired state - not implemented yet - that'll come later.

In k8s we can't track an operation through to an action that k8s takes. All we know is if k8s is happy with the state it's in. We'll be using this in orca after a deploy, update, resize, etc... to decide if the next step is ready. It'll roll up info like health & such. I'll add comments as this code matures.

**Comment by lwander** [comment]
spinnaker/spinnaker#1902

**Comment by lwander** [comment]
spinnaker/spinnaker#2018

**Pull request merged**